### ISSUE TITLE
[Accessibility]: Code intelligence input controls

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35186]
### Audit type

Screen reader navigation

### User journey audit issue

#33516, #33518, #33513

### Problem description

<img width="884" alt="bad use of hints" src="https://user-images.githubusercontent.com/103087/167503588-9595f298-e00e-4e57-954c-0ab82f6bc604.png">

The current label and hint structure around fields makes it possible to navigate but I'm sure we could do much better. It currently reads bare headers without explaining it's a label to an upcoming field, and the "required" hint only comes after the text input.

### Expected behavior

We should figure out what aria settings we are not using or use a different input component entirely. Ensure labels and text fields are linked correctly in the containing form and that we have sufficient hints to describe what each input expects.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by macraig** [comment]
Closing as identified issues have been completed
